src/lib/components/compositions/dialog/dialog.svelte

```diff
@@ -1,27 +1,30 @@
 <script lang="ts">
   import T from '$components/_utilities/locale-strings.svelte';
   import XIcon from '../../icons/x-icon.svelte';
-  import { style } from '$lib/style.store';
+  import { styleStore } from '$lib/style.store';
+  import { componentStore } from '$lib/widget/_utilities/component.utilities';
+
+  import type { ComponentStoreValue } from '$lib/widget/_utilities/component.utilities';
 
-  export let closeCallback: (args: { reason: 'auto' | 'external' | 'user' }) => void;
   export let dialogEl: HTMLDialogElement | null = null;
   export let dialogId: string;
   export let forceOpen = false;
   export let withHeader = false;
 
-  interface CloseOptions {
-    reason: 'auto' | 'external' | 'user';
-  }
-
-  export function closeDialog(args?: CloseOptions) {
-    const { reason } = args || { reason: 'external' };
-
+  // TODO: Add a keyboard listener
+  export function closeDialog(reason: ComponentStoreValue['reason']) {
     function completeClose() {
       dialogEl?.close();
       dialogEl?.classList.remove('tw_dialog-closing');
 
-      // Call dev provided callback function for event hook
-      closeCallback && closeCallback({ reason });
+      // Ensure we have a store and it has an update method on it
+      componentStore?.update((state) => {
```

Comment:
We're directly updating the store here, rather than using the `componentApi`, which as a `close` method, but calling it from here would create a loop in the store, so we directly update the store.